File: windows/src/GoalMaker.Core/Design/Palette.cs
namespace GoalMaker.Core.Design;

/// <summary>
/// One theme's colors for one mode, by role (contracts/design/themes.json, "roles"), as opaque ARGB
/// values so Core stays free of UI types.
/// </summary>
public sealed record Palette(
    uint Background,
    uint Surface,
    uint SurfaceVariant,
    uint Text,
    uint TextMuted,
    uint Outline,
    uint Primary,
    uint OnPrimary,
    uint Accent,
    uint OnAccent,
    uint Hero,
    uint OnHero,
    uint HeroAccent,
    uint Danger);

File: android/app/src/main/kotlin/com/goalmaker/app/domain/design/Palette.kt
package com.goalmaker.app.domain.design

/**
 * One theme's colors for one mode, by role (contracts/design/themes.json, "roles"). Colors are ARGB
 * integers so the domain stays free of UI types.
 */
data class Palette(
    val background: Int,
    val surface: Int,
    val surfaceVariant: Int,
    val text: Int,
    val textMuted: Int,
    val outline: Int,
    val primary: Int,
    val onPrimary: Int,
    val accent: Int,
    val onAccent: Int,
    val hero: Int,
    val onHero: Int,
    val heroAccent: Int,
    val danger: Int,
)
